src/seccomp/mod.rs

```diff
@@ -0,0 +1,423 @@
+use anyhow::bail;
+use anyhow::Context;
+use anyhow::Result;
+use oci_spec::runtime::Arch;
+use oci_spec::runtime::LinuxSeccomp;
+use oci_spec::runtime::LinuxSeccompAction;
+use oci_spec::runtime::LinuxSeccompOperator;
+use seccomp_sys::scmp_arch::*;
+use seccomp_sys::scmp_compare::*;
+use seccomp_sys::*;
+use std::ffi::CString;
+
+#[derive(Debug)]
+struct Compare {
+    // The zero-indexed index of the syscall arguement.
+    arg: libc::c_uint,
+    op: Option<scmp_compare>,
+    datum_a: Option<scmp_datum_t>,
+    datum_b: Option<scmp_datum_t>,
+}
+
+impl Compare {
+    pub fn new(args: u32) -> Self {
+        Compare {
+            arg: args as libc::c_uint,
+            op: None,
+            datum_a: None,
+            datum_b: None,
+        }
+    }
+
+    pub fn op(mut self, op: scmp_compare) -> Self {
+        self.op = Some(op);
+
+        self
+    }
+
+    pub fn datum_a(mut self, datum: scmp_datum_t) -> Self {
+        self.datum_a = Some(datum);
+
+        self
+    }
+
+    pub fn datum_b(mut self, datum: scmp_datum_t) -> Self {
+        self.datum_b = Some(datum);
+
+        self
+    }
+
+    pub fn build(self) -> Result<scmp_arg_cmp> {
+        if self.op.is_some() && self.datum_a.is_some() {
+            Ok(scmp_arg_cmp {
+                arg: self.arg,
+                op: self.op.unwrap(),
+                datum_a: self.datum_a.unwrap(),
+                datum_b: self.datum_b.unwrap_or(0),
+            })
```

Comment:
How about this code? This can remove `unwrap()`.
```suggestion
        if let (Some(op), Some(datum_a)) = (self.op, self.datum_a) {
            Ok(scmp_arg_cmp {
                arg: self.arg,
                op,
                datum_a,
                datum_b: self.datum_b.unwrap_or(0),
            })
```

Reply:
The `scmp_arg_cmp` struct represent a comparison. Some of the `op` require one and some require two value. For example, an equal comparison requires only one value, testing if the syscall argument is equal to this value. In the case where the only one value is needed, I believe the value two or `datum_b` is ignored. The implementation in libseccomp sets the value to 0 anyway, but I don't think it is specifically documented in the man page.